label.js

```diff
@@ -13,33 +13,52 @@ const labelsToRemove = core
 
 async function label() {
   const myToken = core.getInput("repo-token");
+  const ignoreIfAssigned = core.getInput("ignore-if-assigned");
   const octokit = new github.GitHub(myToken);
   const context = github.context;
+  const repoName = context.payload.repository.name;
+  const ownerName = context.payload.repository.owner.login;
+  const issueNumber = context.payload.issue.number;
 
-  let labels = context.payload.issue.labels.map(label => label.name);
+  // query for the most recent information about the issue. Between the issue being created and
+  // the action running, labels or asignees could have been added
+  var updatedIssueInformation = await octokit.issues.get({
+    owner: ownerName,
+    repo: repoName,
+    issue_number: issueNumber
+  });
+
+  if (typeof ignoreIfAssigned !== undefined && ignoreIfAssigned == "True") {
+    // check if the issue has been assigned to anyone
+    if (updatedIssueInformation.data.assignees.length != 0) {
+      return "No action being taken. Ignoring because one or more assignees have been added to the issue";
```

Comment:
Nice!

Reply:
With the latest changes i tested in my private repo and it's working with the following 3 scenarios
- New issue with no labels and no assignee
- New issue with extra labels added
- New issue with extra labels and an assignee added